src/Data/Vec/Functional/Algebra/Base.agda

```diff
@@ -0,0 +1,142 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Some Vector-related module Definitions
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+module Data.Vec.Functional.Algebra.Base where
+
+open import Level using (Level; suc; _⊔_)
+open import Function using (_$_)
+open import Data.Nat using (ℕ)
+open import Data.Fin using (Fin)
+open import Data.Vec.Functional
+open import Algebra.Core
+open import Algebra.Bundles
+open import Algebra.Module
+open import Relation.Binary using (Rel)
+open import Data.Vec.Functional.Relation.Binary.Pointwise using (Pointwise)
+
+private variable
+  a ℓ : Level
+  A : Set ℓ
+  n : ℕ
+
+-- Vector relation lifting
+VecRel : {A : Set a} → Rel A ℓ → Rel (Vector A n) ℓ
+VecRel _≈_ xs ys = Pointwise _≈_ xs ys
+
+-- Binary operation lifting
+lift₂ᴹ : {A : Set a} → Op₂ A → Op₂ (Vector A n)
+lift₂ᴹ _∙_ = zipWith _∙_
+
+-- Unary operation lifting
+lift₁ᴹ : {A : Set a} → Op₁ A → Op₁ (Vector A n)
+lift₁ᴹ f = map f
+
+-- Vector RawMagma construction
+VecRawMagma : RawMagma a ℓ → (n : ℕ) → RawMagma a ℓ
```

Comment:
I would use (say) `Data.Sum.Algebra` as a guide for naming conventions of these.